packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -102,22 +100,24 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
       ? unfilteredNodeMap
       : Object.values(unfilteredNodeMap).filter((node) => searchMatcher(node, searchTerms));
 
-  const rootNodeRefs = Object.values(nodeMap)
+  const unfilteredRootNodeRefs = Object.values(unfilteredNodeMap)
     .filter((item) => !item.parentRef && (!isEnvironmentNode(item) || !isViewing()))
     .map((item) => item.ref);
 
-  const rootNodes: Readonly<ISceneNodeInternal>[] = rootNodeRefs
-    .map(getSceneNodeByRef)
-    .filter((node) => node !== undefined && searchMatcher(node, searchTerms))
-    .map((item) => item as ISceneNodeInternal)
-    .sort(sortNodes);
+  const rootNodes: Readonly<ISceneNodeInternal>[] = !isEmpty(searchTerms)
+    ? Object.values(nodeMap)
+    : unfilteredRootNodeRefs
+        .map(getSceneNodeByRef)
+        .filter((node) => node !== undefined && searchMatcher(node, searchTerms))
+        .map((item) => item as ISceneNodeInternal)
+        .sort(sortNodes);
 
   const getChildNodes = useCallback(
     (parentRef?: string) => {
       const nodeMap = useStore(sceneComposerId).getState().document.nodeMap;
       const results = Object.values(nodeMap)
         .filter((node) => node.parentRef === parentRef)
-        .map(toSceneHeirarchyNode)
+        .map((node) => toSceneHeirarchyNode(node))
```

Comment:
This change creates an unnecessary wrapping function. Super minor, but it's technically another pointer the GC needs to clean up.

Reply:
the map function auto provide a number as second parameter, so if I don't wrap it, there will be error